### PR TITLE
Add basic colcon build test on PR

### PR DESCRIPTION
**Pull request by davidakr**
*Issue #, if available:*

*Description of changes:*
adding GitHub action to build packages via colcon on PR

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: Build packages
+on:
+  pull_request:
+    branches: [ main ]
+  workflow_dispatch:
```
Comment (hartmir) [inline .github/workflows/main.yml:5]: This has no contents - what effect does it have?

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: ros-tooling/setup-ros@v0.6
```
Comment (hartmir) [inline .github/workflows/main.yml:10]: Minor: v0.7 seems latest, any reason to use v0.6?

**Review by hartmir — CHANGES_REQUESTED** [review]
A great addition with minor comments. Would be great to have a more descriptive commit message. We can squash-merge the PR but something meaningful as a base message would be great.

File: .github/workflows/main.yml (hunk 1 above)
Comment (davidakr) [reply .github/workflows/main.yml]: It allows to run the workflow manually. I removed it with the latest commit.

Comment (hartmir) [reply .github/workflows/main.yml]: Running the workflow manually is good IMO - could this go back in please?

Comment (davidakr) [reply .github/workflows/main.yml:5]: I added it in the latest commit.

**Review by hartmir — APPROVED** [review]

**Pull request merged**